cmd/main.go

```diff
@@ -0,0 +1,380 @@
+
+	cliList := []*skupper_ocp_smoke.Client{
+		pubCli,
+		privCli,
+	}
+
+	step := "setup"
+	log.Printf("%s : Starting Setup", step)
+
+	//
+	// Delete the namespace, if set by EnvVar
+	//
+	if strings.ToUpper(os.Getenv("CLEANBEFORE")) == "TRUE" {
+		skupper_ocp_smoke.PrintIfDebug(step, ": Removing namespaces")
+		for _, cli := range cliList {
+			skupper_ocp_smoke.PrintIfDebug(step, ": Remove namespace", cli.Namespace)
+			if err := cli.DeleteNamespace(ctx, step); err != nil {
+				return fmt.Errorf("%s : %s", step, err.Error())
+			}
+			if err := cli.WaitForNSDeletion(ctx, cli.WaitLimit(), step); err != nil {
+				return fmt.Errorf("%s : %s", step, err.Error())
+			}
+		}
+	}
+
+	// Create the namespace
+	log.Printf("%s : Creating namespaces", step)
+	for _, cli := range cliList {
+		skupper_ocp_smoke.PrintIfDebug(step, ": Creating namespace", cli.Namespace)
+		if err := cli.CreateNamespace(ctx, step); err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
+		}
+	}
+
+	// Check if the Skupper operator is available
+	// By default it checks for the operator "skupper-operator" in namespace "openshift-marketplace",
+	// But that can be configured using ENV_VARS
+	if err := pubCli.IsSkupperOperatorAvailable(step, pubCli.OperatorName(), pubCli.OperatorNameSpace(), pubCli.OperatorCatalog()); err != nil {
+		return fmt.Errorf("%s : %s", step, err.Error())
+	}
+
+	// Create the Catalog Groups
+	log.Printf("%s : Create OperatorGroups", step)
+	for _, cli := range cliList {
+		// Create the operatorgroup
+		if err := cli.CreateOperatorGroup(step); err != nil {
+			if !strings.Contains(err.Error(), "\"test-operator-group\" already exists") {
+				return fmt.Errorf("%s : %s", step, err.Error())
+			}
+		}
+		skupper_ocp_smoke.PrintIfDebug(step, ": OperatorGroup created into", cli.Namespace)
+	}
+
+	// Create the subscriptions
+	log.Printf("%s : Create Subscriptions", step)
+	for _, cli := range cliList {
+		if err := cli.CreateSubscription(step); err != nil {
+			if !strings.Contains(err.Error(), "\"skupper-operator\" already exists") {
+				return fmt.Errorf("%s : %s", step, err.Error())
+			}
+		}
+		skupper_ocp_smoke.PrintIfDebug(step, ": subscription created for", cli.Namespace)
+	}
+
+	// Check if subscriptions are active
+	log.Printf("%s : Wait Subscriptions setup", step)
+	for _, cli := range cliList {
+		if err := cli.WaitPodsStatus(ctx, "Running", cli.WaitLimit(), "skupper-site-controller", step); err != nil {
+			return fmt.Errorf("%s : %s", step, err.Error())
+		}
+	}
+
+	// Public Skupper instance
+	log.Printf("%s : Deploy instances", step)
+	pubConfigMapDefinition := v1.ConfigMap{
+		TypeMeta: v12.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v12.ObjectMeta{
+			Name:      "skupper-site",
+			Namespace: pubCli.Namespace,
+		},
+		Immutable: nil,
+		Data: map[string]string{
+			"router-mode":      "interior",
+			"console-user":     "admin",
+			"console-password": "changeme",
+		},
```

Comment:
There is one more issue here. When you run this job on OCP with a Security Context Constraint (scc)
that enforces an uid-range to be provided, the test will fail.

To bypass that I'd suggest you to use something like the Skupper's CLI tests are using, based on the
[getRunAsUserOrDefault](https://github.com/skupperproject/skupper/blob/6a0acae942d8ccb6fe60147196367289c174b761/test/integration/examples/custom/helloworld/main_test.go#L29) function.

With that said, you will need to add two more entries here:
"run-as-user" and "run-as-group", using the recommended ID range.

Reply:
ok, added missing configs